src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/components/ImportFile/index.tsx

```diff
@@ -0,0 +1,207 @@
+  changedMetadata: SampleIdToEditMetadataWebform | null;
+  namedLocations: NamedGisaidLocation[];
+  setMetadata: CommonProps["setMetadata"];
+  hasImportedMetadataFile: boolean;
+  setHasImportedMetadataFile(x: boolean): void;
+  setAutocorrectWarnings(x: SampleIdToWarningMessages): void;
+  setChangedMetadata: CommonProps["setMetadata"];
+}
+
+export default function ImportFile({
+  metadata,
+  namedLocations,
+  setMetadata,
+  setChangedMetadata,
+  hasImportedMetadataFile,
+  setHasImportedMetadataFile,
+  setAutocorrectWarnings,
+  changedMetadata,
+}: Props): JSX.Element {
+  const [missingFields, setMissingFields] = useState<string[] | null>(null);
+  const [autocorrectCount, setAutocorrectCount] = useState<number>(0);
+  const [filename, setFilename] = useState("");
+  const [parseResult, setParseResult] = useState<ParseResult | null>(null);
+  const [extraneousSampleIds, setExtraneousSampleIds] = useState<string[]>([]);
+  const [missingData, setMissingData] =
+    useState<SampleIdToWarningMessages>(EMPTY_OBJECT);
+  const [badFormatData, setBadFormatData] =
+    useState<SampleIdToWarningMessages>(EMPTY_OBJECT);
+
+  useEffect(() => {
+    // If no file uploaded yet, do nothing.
+    if (!parseResult) return;
+    // If file was missing any col header fields, we parsed no data from it
+    // and only display that error to the user to force them to fix problems.
+    if (getMissingFields(parseResult)) return;
+
+    const { data } = parseResult;
+    const parseResultSampleIds = Object.keys(data);
+    const sampleIds = Object.keys(metadata || EMPTY_OBJECT);
+
+    const sampleIdsSet = new Set(sampleIds);
+    const extraneousSampleIds = parseResultSampleIds.filter((parseId) => {
+      return !sampleIdsSet.has(parseId);
+    });
+    setExtraneousSampleIds(extraneousSampleIds);
+  }, [parseResult, metadata]);
+
+  // Used by file upload parser to convert location strings to Locations
+  const stringToLocationFinder = useMemo(() => {
+    return createStringToLocationFinder(namedLocations);
+  }, [namedLocations]);
+
+  const handleFiles = async (files: FileList | null) => {
+    if (!files) return;
+
+    const result = await parseFileEdit(files[0], stringToLocationFinder);
+
+    const { warningMessages, filename } = result;
+    const missingFields = getMissingFields(result);
+    const autocorrectCount =
+      getAutocorrectCount(warningMessages.get(WARNING_CODE.AUTO_CORRECT)) || 0;
+    setHasImportedMetadataFile(true);
+    setMissingFields(missingFields);
+    setAutocorrectCount(autocorrectCount);
+    setFilename(filename);
+    setParseResult(result);
+    setMissingData(
+      warningMessages.get(WARNING_CODE.MISSING_DATA) || EMPTY_OBJECT
+    );
+    setBadFormatData(
+      warningMessages.get(WARNING_CODE.BAD_FORMAT_DATA) || EMPTY_OBJECT
+    );
+
+    handleMetadataFileUpload(result);
+  };
+
+  function inferDeleteEntries(
+    value: string | boolean | NamedGisaidLocation
```

Comment:
I think I saw a few string | boolean | NamedGisaidLocation in a previous PR (not sure if I'm remembering right). If so, maybe consider adding a named type.

Reply:
yeah that's a good idea, i'll address that on my next PR